src/cargo/ops/registry/publish.rs

```diff
@@ -412,13 +414,36 @@ fn transmit(
         return Ok(());
     }
 
+    let deps_map = pkg
+        .dependencies()
+        .iter()
+        .map(|dep| (InternedString::new(dep.name_in_toml().as_str()), dep))
+        .collect::<BTreeMap<InternedString, &Dependency>>();
+
     let string_features = match manifest.original().features() {
         Some(features) => features
             .iter()
             .map(|(feat, values)| {
                 (
                     feat.to_string(),
-                    values.iter().map(|fv| fv.to_string()).collect(),
+                    values
+                        .iter()
+                        .filter(|fv| {
+                            let feature_value = FeatureValue::new(InternedString::new(fv));
+                            match feature_value {
+                                FeatureValue::Dep { dep_name }
+                                | FeatureValue::DepFeature { dep_name, .. } => {
+                                    if let Some(dep) = deps_map.get(&dep_name) {
+                                        dep.is_transitive() || dep.specified_req()
+                                    } else {
+                                        true
+                                    }
```

Comment:
Like we did with the manifest, we should refactor this so we have one way to strip dependencies from a summary and then reuse those results to strip the feature activations, instead of re-implementing the dependency stripping logic in the feature activation code.